vm/systemSmartContracts/delegationManager.go

```diff
@@ -260,11 +260,30 @@ func (d *delegationManager) deployNewContract(
 		return vmcommon.UserError
 	}
 
+	err = d.correctOwnerOnAccount(newAddress, arguments)
+	if err != nil {
+		d.eei.AddReturnMessage(err.Error())
+		return vmcommon.UserError
+	}
+
 	d.eei.Finish(newAddress)
 
 	return vmcommon.Ok
 }
 
+func (d *delegationManager) correctOwnerOnAccount(newAddress []byte, arguments [][]byte) error {
+	if !d.enableEpochsHandler.FixDelegationChangeOwnerOnAccountEnabled() {
+		return nil // backwards compatibility
+	}
+	if len(arguments) == 0 {
+		return vm.ErrInvalidNumOfArguments
+	}
+
+	caller := arguments[0]
```

Comment:
this is wrong. deployNewContract is used in 2 places. When you run from makeNewContractFromValidatorData - arguments[0] == args.CallerAddress. If you run from createNewDelectionContract arguments[0] == maxDelegationCap. Please urse args.CallerAddress here.

Reply:
refactored